lib/hanami/model/sql/types/schema/coercions.rb

```diff
@@ -192,6 +192,12 @@ def self.hash(arg)
                 raise ArgumentError.new("invalid value for Hash(): #{arg.inspect}")
               end
             end
+
+            def self.json(arg)
+              hash(arg)
+            rescue ArgumentError
```

Comment:
This is conceptually wrong and a magnitude slower than checking the current object. Can you please use a code similar to this? https://github.com/hanami/model/pull/422/files#diff-92790045433e0619f41285e4ad4b4af0R34 Thanks!